packages/editor-ui/src/stores/assistant.store.ts

```diff
@@ -63,6 +63,10 @@ export const useAssistantStore = defineStore(STORES.ASSISTANT, () => {
 	const currentSessionActiveExecutionId = ref<string | undefined>();
 	const currentSessionWorkflowId = ref<string | undefined>();
 	const lastUnread = ref<ChatUI.AssistantMessage | undefined>();
+	const nodeExecutionStatus = ref<'not_executed' | 'success' | 'error'>('not_executed');
+	// This is used to show a message when the assistant is performing intermediate steps
+	// We use streaming for assistants that support it, and this for agents
+	const assistantThinkingMessage = ref<string | undefined>();
```

Comment:
why could not this be pushed to the same messages array? why does it need to break the design pattern here?

Reply:
We wont be able to animate the transitions then since the whole message list is re-rendered as messages come from the service.
Also, in my opinion, this is not a message from assistant. it's a thinking state so I think it makes sense to have a separate slot for it here

Reply:
Would not adding them and removing from the array with different ids have the same effect?

Reply:
Will try that, but I don't think it will look the way we want it. The effect we are after is that assistant icon should stay in place (pulsating) while the message text should rotate as steps change

Reply:
No need. I don't want to create unnecessary work. If we think of it as a loading state it makes sense.